src/edu/csus/ecs/pc2/shadow/ShadowScoreboardComparator.java

```diff
@@ -0,0 +1,391 @@
+        
+        String teamsJson = "";
+        Map<String,String> teamNameMap ;
+        
+        switch (scoreboardType) {
+            
+            case PC2:
+                //get the CLICS JSON describing the PC2 teams 
+                teamsJson = shadowController.getPC2TeamsJSON();
+                
+                break;
+                
+            case REMOTE:
+                //get the CLICS JSON describing the remote teams 
+                teamsJson = shadowController.getRemoteTeamsJSON();
+                
+                break;
+                
+            default:
+                log.severe("Unknown scoreboard type: " + scoreboardType);
+                return;
+        }
+        
+        
+        //create a Map which maps team ids to team names
+        teamNameMap = createTeamNameMap(teamsJson);
+        
+        //process every rank in the received map (which maps ranks to PriorityQueues containing TeamScoreRows holding that rank)
+        for (int rank : rowMap.keySet()) {
+            
+            PriorityQueue<TeamScoreRow> pq = rowMap.get(rank);
+            
+            for (TeamScoreRow row : pq) {  //iterates over all TeamScoreRows for the current rank, although not necessarily in priority order
+                
+                String teamIdAsString = new Integer(row.getTeam_id()).toString();
+                
+                row.setTeamName(teamNameMap.get(teamIdAsString));
+                
+            }
+        }
+        
+    }
+
+
+    /**
+     * Converts the specified Teams JSON string to a map which maps teamId to team name.
+     * 
+     * @param teamsJson a JSON string representing teams, in CLICS format
+     * 
+     * @return a Map which maps team numbers (Ids) to corresponding team names.
+     *          The returned map may be empty (if for example the received teamsJson string
+     *          is null or empty, or if an exception occurs while processing the teamsJson string),
+     *          but the returned map will never be null.
+     */
+    private Map<String, String> createTeamNameMap(String teamsJson) {
+        
+        Map<String,String> retMap = new HashMap<String,String>();
+        
+        //insure we have potentially reasonable input
+        if (teamsJson==null || teamsJson.contentEquals("")) {
+            log.severe("Null or empty teams JSON");
+            return retMap;
+        }
+        
+        //convert the provided teamsJson, assumed to be a CLICS teams description, into a POJO
+        // containing a list of teams
+        ObjectMapper mapper = new ObjectMapper();
+        
+        List<Team> teamList = null;
+        try {
+            teamList = mapper.readValue(teamsJson, new TypeReference<List<Team>>() {});
+        } catch (IOException e) {
+            log.severe("Exception converting teams JSON description to TeamList class: " + e.getMessage());
```

Comment:
Add the exception to the log message.   s/e.getMessage()/e.getMessage(), e);
If it is severe we want the stack trace in the log.

Reply:
The suggested change will not work; method Log.severe() does not support passing a second parameter of type Exception.  However, I have made an equivalent change so that the exception gets logged.  I also removed the "debug output" which follows the log statement.